### PR TITLE
Initial service part 2

### PR DESCRIPTION
**Pull request by ianmcorvidae**
This adds an update to force an update of usage information, adds an AMQP listener and publisher, adds a single-user update via AMQP, and publishes a single-user update message via AMQP when a user requests their data usage information and it's older than the configured refresh interval.

### CODE REVIEW

**Comment by ianmcorvidae** [comment]
I also reorganize a bunch of stuff, so, heh, sorry about that.

File: main.go [hunk 1]
```diff
@@ -53,88 +77,71 @@ func main() {
...
+		configuration.AMQPExchangeType,
+		queueName,
+		[]string{"index.all", "index.usage.data", "index.usage.data.batch.user.#", a.SingleUserPrefix + ".#"},
+		func(del amqp.Delivery) {
+			var err error
+			log.Tracef("Got message: %s", del.RoutingKey)
+			if del.RoutingKey == "index.all" || del.RoutingKey == "index.usage.data" {
+				// generate prefixes and publish
+			} else if strings.HasPrefix(del.RoutingKey, "index.usage.data.batch.user") {
+				// handle batch
```
Comment (ash3rz) [inline main.go:127]: Is this condition and the condition above going to be handled in a later PR, or maybe I missed some other way of handling these 2 cases?

**Review by ash3rz — APPROVED** [review]
(not a Go expert) I had one small question, but otherwise didn't see any showstoppers 🚢

File: main.go (hunk 1 above)
Comment (ianmcorvidae) [reply main.go:127]: Yeah, these are for later PRs. The first one will fetch all usernames from the DB(s) and publish a bunch of `index.usage.data.batch.user.*` messages. The second one will consume one of those messages and update the data usage for everything within the batch (or possibly just everything whose current info is older than the refresh interval). But, not done yet. For now it'll just acknowledge the messages and carry on.

**Review by psarando — APPROVED** [review]
No Go expert myself, but I didn't catch any issues 👍

**Comment by ianmcorvidae** [comment]
Alright, I'll go with two non-expert reviews as good to merge, though more review is of course welcome. Thanks for giving it a look!

**Pull request merged**